### PR TITLE
Merging changes from dev to master

### PR DESCRIPTION
**Pull request by ayush9pandey**
Multiple people have contributed to dev over the last couple years. Let's make sure that all of these commits look good and then we can merge. Requesting @sclamons @WilliamIX to review.

### CODE REVIEW

**Comment by ayush9pandey** [comment]
@WilliamIX  could you review this PR so we can merge? This is just dev to master - so I believe this code has been vetted once when we merged to dev? I want to send in a couple PRs after this.

File: bioscrape/inference.pyx [hunk 1]
```diff
@@ -256,18 +321,28 @@ cdef class ModelLikelihood(Likelihood):
             for (key,val) in  pairs:
                 self.init_state_indices[index] = self.m.get_species_index( key  )
                 self.init_state_vals[index] = val
-                index +=  1
+                index +=  1"""
```
Comment (WilliamIX) [inline bioscrape/inference.pyx:324]: Should these """ be here?

Comment (ayush9pandey) [reply bioscrape/inference.pyx]: yeah, it's the multiline comment end. git diff detected it incorrectly. I put these """ in a new line so it is clearer.

Comment (WilliamIX) [reply bioscrape/inference.pyx]: Great. Otherwise Looks good.

**Review by WilliamIX — APPROVED** [review]

**Comment by ayush9pandey** [comment]
Thanks! Merging.

**Pull request merged**